bundles/org.openhab.binding.ferroamp/src/main/java/org/openhab/binding/ferroamp/internal/FerroampChannelConfiguration.java

```diff
@@ -0,0 +1,365 @@
+/**
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.ferroamp.internal;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.measure.Unit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.core.library.unit.Units;
+
+/**
+ * The {@link FerroampChannelConfiguration} class defines methods, that set the channel configuration for the binding.
+ *
+ * @author Örjan Backsell - Initial contribution
+ *
+ */
+
+@NonNullByDefault
+public class FerroampChannelConfiguration {
+    public String id;
+    public Unit<?> unit;
+
+    public FerroampChannelConfiguration(String id, Unit<?> unit) {
+        this.id = id;
+        this.unit = unit;
+    }
+
+    public static List<FerroampChannelConfiguration> getChannelConfigurationEhub() {
+        final List<FerroampChannelConfiguration> channelConfigurationEhub = new ArrayList<>();
+        channelConfigurationEhub
+                .add(new FerroampChannelConfiguration(FerroampBindingConstants.CHANNEL_GRIDFREQUENCY, Units.HERTZ));
```

Comment:
By adding a private helper and some minor refactoring the `getChannelConfigurationEhub` can be made much more readable.
Obiously the same for `getChannelConfigurationSsoS1` and the other methods in this class.

```suggestion

    private static FerroampChannelConfiguration cc(String id, Unit<?> unit) {
        return new FerroampChannelConfiguration(id, unit);
    }

    public static List<FerroampChannelConfiguration> getChannelConfigurationEhub() {
        final List<FerroampChannelConfiguration> list = new ArrayList<>();
        list.add(cc(FerroampBindingConstants.CHANNEL_GRIDFREQUENCY, Units.HERTZ));
```

Reply:
Done as described for all methods in class.

Reply:
Much better!